src/core/modules/impl/handler/AbstractContractHandler.ts

```diff
@@ -142,7 +142,13 @@ export abstract class AbstractContractHandler<State> implements HandlerApi<State
           const lastErrorKey = errorKeys[errorKeys.length - 1] as string;
           const lastErrorMessage = stateWithValidity?.cachedValue?.errorMessages[lastErrorKey];
           // don't judge me..
-          if (lastErrorMessage.startsWith('[SkipUnsafeError]')) {
+          // FIXME: also - '?' is stinky...
+          if (
+            lastErrorMessage &&
+            lastErrorMessage.startsWith &&
+            (lastErrorMessage.startsWith('[SkipUnsafeError]') ||
+              lastErrorMessage.startsWith('[NonWhitelistedSourceError]'))
```

Comment:
If the name `SkipUnsafeError` is part of the error handling logic maybe it should have some dedicated type, so we can easily search the code.
Also how about 
```javascript
lastErrorMessage.startsWith(`[${KnownErrors.NonWhitelistedSourceError}]`))
```